src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+		}
+	}
+
+	@Override
+	public void transform(CmsItem item, TransformConfig config) {
+		if (config == null || config.getOptions() == null) {
+			throw new IllegalArgumentException("TransformServiceXsl needs a valid TransformConfig object.");
+		}
+		
+		final CmsItem base = item;
+		final CmsItemId baseItemId = base.getId();
+		final CmsRepository repository = baseItemId.getRepository();
+		final RepoRevision baseRevision = repoLookup.getYoungest(repository);
+		final Boolean overwrite = new Boolean(config.getOptions().getParams().get("overwrite"));
+		
+		if (!config.getOptions().getType().equals("xsl")) {
+			throw new IllegalArgumentException("TransformServiceXsl can only handle xsl transforms but was given: " + config.getOptions().getType());
+		}
+		
+		final String stylesheet = config.getOptions().getParams().get("stylesheet");
+		if (stylesheet == null || stylesheet.trim().isEmpty()) {
+			throw new IllegalArgumentException("Requires a valid stylesheet path or stylesheet name.");
+		}
+		
+		final CmsItemPath outputPath = getOutputPath(baseItemId ,config.getOptions().getParams().get("output"));
+		if (!pathExists(repository, outputPath)) {
+			throw new IllegalArgumentException("Specified output must be an existing folder: " + outputPath.getPath());
+		}
+		
+		final TransformerService transformerService = getTransformerService(baseItemId, stylesheet);
+		if (transformerService == null) {
+			throw new IllegalArgumentException("Could not create transformerService with stylesheet: " + stylesheet);
+		}
+		
+		transformerService.setItemLookup(itemLookup);
+		
+		final CmsPatchset patchset = new CmsPatchset(repository, baseRevision);
+		patchset.setHistoryMessage(config.getOptions().getParams().get("comment"));
+		
+		final CmsItemPropertiesMap props = getProperties(base, config);
+		
+		SaxonOutputURIResolverXdm outputURIResolver = new SaxonOutputURIResolverXdm(sourceReader);
+		TransformOptions transformOptions = new TransformOptions();
+		transformOptions.setOutputURIResolver(outputURIResolver);
+		
+		TransformStreamProvider baseStreamProvider = transformerService.getTransformStreamProvider(baseItemId, transformOptions);
+		addToPatchset(patchset, outputPath.append(baseItemId.getRelPath().getName()), baseStreamProvider, overwrite, props);
+		
+		Set<String> resultDocsHrefs = outputURIResolver.getResultDocumentHrefs();
+		for (String relpath: resultDocsHrefs) {
+			if (relpath.startsWith("/")) {
+				throw new IllegalArgumentException("Relative href must not start with slash: " + relpath);
+			}
+			
+			XmlSourceDocumentS9api resultDocument = outputURIResolver.getResultDocument(relpath);
+			TransformStreamProvider streamProvider = transformerIdentity.getTransformStreamProvider(resultDocument, null);
+			
+			CmsItemPath path = outputPath.append(Arrays.asList(relpath.split("/")));
+			addToPatchset(patchset, path, streamProvider, overwrite, props);
+		}
+		
+		commit.run(patchset);
```

Comment:
Do we have a strategy for error handling? Should we log here? 

I think ItemEventCommandHandler should wrap each listener in try.